Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -221,6 +221,7 @@ CachedResourceLoader::~CachedResourceLoader()
 CachedResource* CachedResourceLoader::cachedResource(const String& resourceURL) const
 {
     ASSERT(!resourceURL.isNull());
+    ASSERT(m_document);
     return cachedResource(MemoryCache::removeFragmentIdentifierIfNeeded(m_document->completeURL(resourceURL)));
```

Comment:
If `m_document` could be `nullptr` at any time, we should be doing more than just asserting.

Would like to see something more like this (maybe even remove the `ASSERT(m_document);` statement:

```suggestion
    ASSERT(m_document);
    return m_document ? cachedResource(MemoryCache::removeFragmentIdentifierIfNeeded(m_document->completeURL(resourceURL))) : nullptr;
```